proposals/explicit-backing-fields.md

```diff
@@ -14,417 +14,396 @@ but one is part of a public API and another is an implementation detail.
+    
+    fun updateCity(newCity: String) {
+        city.value = newCity // visible as MutableLiveData, calling field
+    }
+}
 
-Backing field assignability is the same as it is now for `field` references in getters and setters:
+fun outside(vm: SomeViewModel) {
+    vm.city // visible as LiveData, calling getter
+} 
+```
 
-* `var` properties have mutable backing fields.
-* `val` properties have read-only backing fields. That is, assignment to `field` inside a `val` property getter results in an error.
+There is no possibility to call a getter inside the explicit backing field visibility scope
+(might be reconsidered in [Future enhancements](#future-enhancements)).
+
+### Accessors
+
+Property, which has explicit backing field, can also specify getter or setter.
+
+```kotlin
+val counter: Int
+    field = atomic(0)
+    get() = field.value
+```
 
-#### Accessors
-                                                          
-When explicit backing field with type `F` for a property with type `P` is declared explicitly, 
+When explicit backing field with type `F` for a property with type `P` is declared,
 then compiler can derive getter and, for `var` properties, setter implementation automatically:
 
 * If `P :> F`, the compiler can provide a default getter.
 * If `P <: F`, the compiler can provide a default setter.
 
-If the compiler can not provide a getter, the user must declare it explicitly. 
+If the compiler can not provide a getter, the user must declare it explicitly.
 The same applies to setters in case of `var` properties.
 
 ```kotlin
-public val flow: SharedFlow<String>
-    field: MutableSharedFlow? = null
-    get() { // It is an error if getter is not explicitly specified here
-        return field ?: run { ... }
-    }
+val city: String
+    field = MutableLiveData<String>()
+    get() = field.value ?: "-" // It is an error if getter is not explicitly specified here
 ```
 
-#### Visibility
+### Explicit backing field in combination with property initializer
 
-Only the `private` and `internal` visibilities are allowed for explicit backing fields. 
-The default field visibility is `private`.
+It's allowed to have both property initializer and explicit backing field specified.
 
 ```kotlin
-val mutableWithinModule: List<Item>
-    internal field = mutableListOf()
+val city: StateFlow<String> = field.asStateFlow()
+    field = MutableStateFlow("")
 ```
 
-The special syntax to explicitly access the backing field from outside the code of property accessors is TBD, 
-but the field can be implicitly access when it is visible via the [Smart Type Narrowing](#smart-type-narrowing).
+In this case, the property initializer expression (`field.asStateFlow()` in example above) is evaluated once
+immediately after the backing field is initialized and then returned every time the getter of property is called.
 
-#### Lateinit
+If property with explicit backing field has initializer, it's prohibited to declare accessors.
```

Comment:
One more candidate for the future enhancements list? Or a way "to prevent feature abuse"?